src/pinecone.ts

```diff
@@ -312,15 +310,13 @@ export class Pinecone {
    * @returns A promise that resolves when the request to delete the index is completed.
    * @throws {@link Errors.PineconeArgumentError} when invalid arguments are provided
    */
-  deleteIndex(indexName: IndexName) {
-    const deleteIndexPromise = this._deleteIndex(indexName);
+  async deleteIndex(indexName: IndexName) {
```

Comment:
Same as above but for evicting the host from `IndexHostSingleton` after a successful delete.